### DISCUSSION TITLE
Applying images as texture on a mesh

### DISCUSSION THREAD

**Discussion by sharoseali** [pyvista/pyvista#1814] [Q&A] [answered]
@banesullivan @akaszynski
I have to apply a texture on a mesh, but the code doesn't seem to be working.  [Here are the details](https://github.com/pyvista/pyvista-support/issues/159#issuecomment-970813699). Thanks

**Comment by banesullivan** [comment, marked as answer]
The data you shared over in https://github.com/pyvista/pyvista-support/issues/159#issuecomment-970813699 doesn't much look like the zebra in those pictures:

```py
import numpy as np
import pyvista as pv

pts = np.loadtxt('/Users/bane/Downloads/recovered_10.xyz')

pc = pv.PolyData(pts[:, 0:3])
pc['vecs'] = pts[:, 3:]

pc.plot(notebook=0, color=True, eye_dome_lighting=True)
```
| data | your post |
|---|---|
| <img width="624" alt="Screen Shot 2021-11-26 at 12 57 46 AM" src="https://user-images.githubusercontent.com/22067021/143546786-42ceb982-342d-46a8-8c96-e2acfb84f051.png"> |  <img width="760" alt="Screen Shot 2021-11-26 at 1 02 12 AM" src="https://user-images.githubusercontent.com/22067021/143547239-aca85ff0-3c50-4d3e-8a59-fa1c99b4bc6e.png">|

-----

Beyond that... I'm closing this because this is out of scope of what we can help you with. You'll need to process these data these data in a few ways that are out of scope for PyVista including (even if you have the correct point cloud of the zebra):

1. Perform some sort of meshing of the point cloud to generate a mesh with surface topology
2. Perform the actual texture mapping of the images onto that surface (this is the hardest, good luck)

After that, PyVista can help you to simply visualize that data. Otherwise, I don't think we can help you solve this rather complex task

**Reply by sharoseali** [reply]
Thanks for your answer. Sorry I share the wrong point cloud with you.  My workflow is to get raw point cloud, apply some cleaning and make it packed mesh then finally apply texture on it from the corresponding image, but the problem arrives after cleaning the mesh and texture mapping code didn't work. I was also thinking to crop zebra from the image or remove the background to map it on a mesh. But i don't know the exact way out (I just stepped into 3D world). [Here](https://drive.google.com/file/d/153Fy8UtWnM7ooAhAEsVwU6rDhwpHhqKu/view?usp=sharing) is the correct mesh. thanks again
